app/controllers/api_users_controller.rb

```diff
@@ -28,15 +29,19 @@ def create
   def post_to_endpoint(user)
     @user = { email: user.email, department: user.department, service: user.service }
     uri = URI.parse(Rails.configuration.self_service_api_endpoint)
-    http = Net::HTTP.new(uri.host, uri.port)
-    http.use_ssl = (uri.scheme == 'https')
-
-    http.start do |http_start|
-      request = Net::HTTP::Post.new(uri.request_uri)
-      request.basic_auth(ENV['SELF_SERVICE_HTTP_AUTH_USERNAME'], ENV['SELF_SERVICE_HTTP_AUTH_PASSWORD'])
-      request.set_form_data(@user)
-      http_start.request(request)
+    options = {
+      basic_auth: { username: ENV['SELF_SERVICE_HTTP_AUTH_USERNAME'], password: ENV['SELF_SERVICE_HTTP_AUTH_PASSWORD'] },
```

Comment:
It would be a good idea to check that both username and password ENV are not `nil`. Ideally in the boundaries of the application. Perhaps a good initial step is to change to `ENV.fetch('SELF_SERVICE_HTTP_AUTH_USERNAME')` so at least the error will be very clear on what is wrong.

Reply:
Also, to keep the docker deploy working we should probably have a `docker-compose.yml` on the lines of:

```yaml
version: '3'
services:
  db:
    image: postgres:9.6-alpine
    volumes:
      - data:/var/lib/postgresql/data/pgdata
    environment:
      POSTGRES_PASSWORD:
      POSTGRES_USER: openregister-info
      POSTGRES_DB: openregister-info_development
      PGDATA: /var/lib/postgresql/data/pgdata
  web:
    build: .
    ports:
      - "3000:3000"
    depends_on:
      - db
    environment:
      SELF_SERVICE_HTTP_AUTH_USERNAME: ${SELF_SERVICE_HTTP_AUTH_USERNAME}
      SELF_SERVICE_HTTP_AUTH_PASSWORD: ${SELF_SERVICE_HTTP_AUTH_PASSWORD}

volumes:
  data: {}
```